rskj-core/src/main/java/co/rsk/config/mining/StableMinGasPriceSystemConfig.java

```diff
@@ -0,0 +1,50 @@
+package co.rsk.config.mining;
+
+import co.rsk.mine.gas.MinGasPriceProviderType;
+import com.typesafe.config.Config;
+
+public class StableMinGasPriceSystemConfig {
+    public static final String STABLE_GAS_PRICE_CONFIG_PATH_PROPERTY = "miner.stableGasPrice";
+    private static final String ENABLED_PROPERTY = "enabled";
+    private static final String REFRESH_RATE_PROPERTY = "refreshRate";
+    private static final String MIN_STABLE_GAS_PRICE_PROPERTY = "minStableGasPrice";
+    private static final String METHOD_PROPERTY = "method";
+
+    private final Integer refreshRate;
+    private final Integer minStableGasPrice;
+    private final boolean enabled;
+    private final MinGasPriceProviderType method;
+    private final Config config;
+
+    public StableMinGasPriceSystemConfig(Config config) {
+        enabled = config.getBoolean(ENABLED_PROPERTY);
+        refreshRate = config.getInt(REFRESH_RATE_PROPERTY);
+        minStableGasPrice = config.getInt(MIN_STABLE_GAS_PRICE_PROPERTY);
+        method = config.getEnum(MinGasPriceProviderType.class, METHOD_PROPERTY);
```

Comment:
If you agree with the change on structure of the config file, you would have to change a bit this logic and read the source object, then convert it to an object with the method and the key, value map. Something like this:

```
public class StableSourceGasPrice {

    private final MinGasPriceProviderType method;
    private final Map<String,String> params;
...
}
```

Then this method would be filled using the enum function converion, like this:

```
        method = MinGasPriceProviderType.fromMethodType(config.getString(METHOD_PROPERTY));
```

But I am not sure if it's possible to read the config like the object and extract the map. I am assuming it is, let me know your thoughts. 🙂